pkg/auth/service.go

```diff
@@ -959,3 +964,626 @@ func (s *DBAuthService) Authorize(ctx context.Context, req *AuthorizationRequest
 	// we're allowed!
 	return &AuthorizationResponse{Allowed: true}, nil
 }
+
+type APIAuthService struct {
+	apiClient   *ClientWithResponses
+	secretStore crypt.SecretStore
+	cache       Cache
+}
+
+func (a *APIAuthService) SecretStore() crypt.SecretStore {
+	return a.secretStore
+}
+
+func (a *APIAuthService) CreateUser(ctx context.Context, user *model.User) (int, error) {
+	resp, err := a.apiClient.CreateUserWithResponse(ctx, CreateUserJSONRequestBody{
+		Username: user.Username,
+	})
+	if err != nil {
+		return InvalidUserID, err
+	}
+	if err := expectedStatusCode(resp, 201); err != nil {
+		return InvalidUserID, err
+	}
+
+	return resp.JSON201.Id, nil
+}
+
+func (a *APIAuthService) DeleteUser(ctx context.Context, username string) error {
+	u, err := a.GetUser(ctx, username)
```

Comment:
why do we need to get the user first - this is not the remote implementation's work?

Reply:
delete user is the only one that works with ID - the service can delete user by name